osversion/platform_compat_windows.go

```diff
@@ -0,0 +1,70 @@
+package osversion
+
+import (
+	"strconv"
+	"strings"
+)
+
+// List of stable ABI compliant
+// ltsc releases
+// Note: List must be sorted
+var compatLtscReleases = []uint{
+	V21H2Server,
+}
+
+// Checks for stable ABI compliant OS versions.
+// Every release after WS 2022 will support the previous
+// ltsc container image. Stable ABI is in preview mode
+// for windows 11 client.
+// Refer: https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2022%2Cwindows-10#windows-server-host-os-compatibility
+func CanRun(hostOSVersion, ctrImgVersion string) bool {
+	// HPC containers do not have the hostOSVersion set
+	if hostOSVersion == "" {
+		return true
+	}
+
+	hostOSBuildNum := GetOSBuildNumber(hostOSVersion)
+	ctrImgBuildNum := GetOSBuildNumber(ctrImgVersion)
+
+	// If host is < WS 2022, exact version match is required
+	if hostOSBuildNum < V21H2Server {
+		return strings.HasPrefix(ctrImgVersion, hostOSVersion)
+	}
+
+	// if host is >= WS2022, check for stable ABI compliance
+	hostOSMajorMinorVersion := GetMajorMinorVersion(hostOSVersion)
+	ctrImgMajorMinorVersion := GetMajorMinorVersion(ctrImgVersion)
+	if hostOSMajorMinorVersion != ctrImgMajorMinorVersion {
+		return false
+	}
+
+	var supportedLtscRelease uint
+	for i := len(compatLtscReleases) - 1; i >= 0; i-- {
+		if hostOSBuildNum >= compatLtscReleases[i] {
+			supportedLtscRelease = compatLtscReleases[i]
+			break
+		}
+	}
+	return ctrImgBuildNum >= supportedLtscRelease && ctrImgBuildNum <= hostOSBuildNum
+}
+
+func GetOSBuildNumber(v string) uint {
+	parts := strings.Split(v, ".")
+	if len(parts) < 3 {
+		return 0
+	}
+
+	r, err := strconv.Atoi(parts[2])
+	if err != nil {
+		return 0
+	}
+	return uint(r)
+}
+
+func GetMajorMinorVersion(v string) string {
```

Comment:
same here?

Reply:
I thought it could be a useful function to have.
This function is currently part of containerd/platforms/default_windows.go function and I thought having windows specific functions to parse the build version from a given OS version string would be more meaningful to have in osversion package of hcsshim instead of containerd.